### DISCUSSION TITLE
FEAT: prioritize exact matches on param name for intellisense tab-completions

### DISCUSSION THREAD

**Discussion by NickCrews** [microsoft/pylance-release#7122] [Enhancement]
Given:
- I have a function `def foo(haystack: str, needle: str)`
- at a call site I begin typing `foo(haystack=bar, needle=nee`
- In scope are several variables:
   - needed_columns_h
   - needed_columns_n
   - needle
 
Then intellisense suggests all three of these to me. But it sorts them alphabetically. So this leads to it suggesting `needed_columns_h` as the first choice. I would want `needle` to be the first choice.

I want to avoid the whole can of worms for more complex rankings of "which in-scope variable is the most likely to be correct", so I think even if it only prioritizes exact matches, that would be great. Slightly more complex, and probably still a better experience would be if it ranked them by eg damerau-levenshtein distance, and then broke ties with alphabetical sorting.

![Image](https://github.com/user-attachments/assets/fc426d21-e861-49b0-b2bb-89e768493b92)

**Comment by debonte** [comment]
Moving this issue to discussions as an enhancement request for comments and upvotes.
